x-pack/plugins/fleet/server/saved_objects/index.ts

```diff
@@ -156,6 +157,7 @@ export const getSavedObjectTypes = (): { [key: string]: SavedObjectsType } => ({
         is_protected: { type: 'boolean' },
         overrides: { type: 'flattened', index: false },
         keep_monitoring_alive: { type: 'boolean' },
+        ...getSettingsSavedObjectMappings('AGENT_POLICY_ADVANCED_SETTINGS'),
```

Comment:
> Saved objects
*I think we will still have to do some work to add a new model version when adding a new saved object field, I do not see an easy way to programatically generate that. In a first time it probably could be a manual action to add those migration

I see what you mean, without adding model versions, I am getting an error when trying to set the new field
```
{
    "statusCode": 400,
    "error": "Bad Request",
    "message": "[1:349] mapping set to strict, dynamic introduction of [agent_limits_go_max_procs] within [ingest-agent-policies] is not allowed: strict_dynamic_mapping_exception\n\tRoot causes:\n\t\tstrict_dynamic_mapping_exception: [1:349] mapping set to strict, dynamic introduction of [agent_limits_go_max_procs] within [ingest-agent-policies] is not allowed"
}
```
Added this `modelVersions` locally to get it working:
```
    modelVersions: {
      '1': {
        changes: [
          {
            type: 'mappings_addition',
            addedMappings: {
              ...getSettingsSavedObjectMappings('AGENT_POLICY_ADVANCED_SETTINGS'),
            },
          },
        ],
      },
    },
```

I am wondering if we could add a generic mapping here, since these fields are usually not indexed anyway, something like:
```
advanced_settings: { type: 'object', index: false }
```

Reply:
I made this work locally, to save these settings under a generic `advanced_settings` mapping in the SO. The full agent policy hasn't changed.
```
PUT kbn:/api/fleet/agent_policies/b180cbb0-b276-4f96-97a7-e2acb97c6a36
{
  "name": "Agent policy 2",
  "description": "",
  "namespace": "default",
  "monitoring_enabled": [
    "logs",
    "metrics"
  ],
  "inactivity_timeout": 1209600,
  "is_protected": false,
  "advanced_settings": {
    "agent_limits_go_max_procs": 6
  }
}
```